src/main/java/frc/robot/subsystems/IndexerSubsystem.java

```diff
@@ -33,48 +41,50 @@ public void activateConveyor() {
     /**
      * Turn off the conveyor
      */
-    public void deactivateConveyor() {
+    public void stopConveyor() {
         conveyorMotor.stopMotor();
     }
 
     /**
-     * Activate or disable the conveyor
-     * 
-     * @param on If the conveyor should be on or off
+     * Activate the gate wheel at the default speed for indexing a cargo
      */
-    public void setConveyor(boolean on) {
-        if (on) {
-            activateConveyor();
-        } else {
-            deactivateConveyor();
-        }
+    public void activateGateWheelForIndexing() {
+        gateMotor.set(ControlMode.PercentOutput, GATE_WHEEL_INDEXING_POWER);
     }
 
     /**
-     * Activate the gate wheel at defult speed
+     * Activate the gate wheel at the default speed for shooting
      */
-    public void activateGateWheel() {
-        gateMotor.set(ControlMode.PercentOutput, CONVEYOR_POWER);
+    public void activateGateWheelForShooting() {
+        gateMotor.set(ControlMode.PercentOutput, GATE_WHEEL_SHOOTING_POWER);
     }
 
     /**
      * Turn off the gate wheel
      */
-    public void deactivateGateWheel() {
-        conveyorMotor.stopMotor();
+    public void stopGateWheel() {
+        gateMotor.stopMotor();
     }
 
     /**
-     * Activate or disable the gate wheel
+     * Roll the gate wheel by the specified amount
      * 
-     * @param on If the gete wheel should be on or off
+     * @param degrees The distance to move the wheel
+     */
+    public void rotateGateWheelByXDegrees(double degrees) {
+        double targetPosition = getGateWheelPosition(degrees) + degrees;
+        gateMotor.set(ControlMode.Position, UnitConversions.Indexer.gateWheelDegreesToEncoderTicks(targetPosition));
```

Comment:
This function name is a bit confusing as we aren't rotating the gate wheel so much as telling it a target position to try and achieve. I would suggest renaming to be a bit more clear.

Secondarily, the moment the gateMotor is set to ControlMode.Position we are assuming that a cargo is loaded even though it could still be trying to index the cargo to prepare to fire. I would also suggest having a more robust way to know when the cargo is ready or now (only have the cargo be considered loaded when this function ends / the motor gets to the position?)

Reply:
Again, I don't really have a reliable way to detect the presence of a cargo without a beam-break sensor.

You are right, using the closed loop error to tell if a cargo is indexed is probably a much better idea than simply what mode the motor controller is in. I'll implement that.